src/Glpi/Form/QuestionType/QuestionTypeUserDevice.php

```diff
@@ -0,0 +1,268 @@
+                        value="1" {{ is_multiple_devices ? 'checked' : '' }}
+                        onchange="handleMultipleDevicesCheckbox_{{ rand }}(this)"
+                        data-glpi-form-editor-specific-question-extra-data>
+                    <span class="form-check-label">{{ is_multiple_devices_label }}</span>
+                </label>
+            </div>
+
+            <script>
+                function handleMultipleDevicesCheckbox_{{ rand }}(input) {
+                    const is_checked = $(input).is(':checked');
+                    const selects = $(input).closest('section[data-glpi-form-editor-question]')
+                        .find('div .devices-dropdown');
+
+                    {# Toglle all selects visibility #}
+                    selects.toggleClass('d-none');
+
+                    {# Handle hidden input for multiple devices #}
+                    selects.find('input[type="hidden"]').prop('disabled', !is_checked);
+                }
+            </script>
+TWIG;
+
+        $twig = TemplateRenderer::getInstance();
+        return $twig->renderFromStringTemplate($template, [
+            'is_multiple_devices' => $this->isMultipleDevices($question),
+            'is_multiple_devices_label' => __('Allow multiple devices')
+        ]);
+    }
+
+    #[Override]
+    public function renderEndUserTemplate(Question $question): string
+    {
+        $template = <<<TWIG
+            {% import 'components/form/fields_macros.html.twig' as fields %}
+
+            {{ fields.dropdownArrayField(
+                question.getEndUserInputName(),
+                '',
+                items,
+                '',
+                {
+                    'no_label'           : true,
+                    'field_class'        : 'col-12',
+                    'display_emptychoice': true,
+                    'multiple'           : is_multiple_devices,
+                    'aria_label'         : aria_label,
+                }
+            ) }}
+TWIG;
+
+        $twig = TemplateRenderer::getInstance();
+        return $twig->renderFromStringTemplate($template, [
+            'question' => $question,
+            'items'    => CommonItilObject_Item::getMyDevices(Session::getLoginUserID(), Session::getActiveEntities()),
+            'is_multiple_devices' => $this->isMultipleDevices($question),
+            'aria_label' => _n('Select device...', 'Select devices...', $this->isMultipleDevices($question) ? 2 : 1),
+        ]);
+    }
+
+    #[Override]
+    public function prepareEndUserAnswer(Question $question, mixed $answer): mixed
+    {
+        if (!is_array($answer)) {
+            $answer = [$answer];
+        }
+
+        $devices = [];
+        if (is_array($answer)) {
+            foreach ($answer as $device) {
+                $device_parts = preg_split('~_(?=[^_]*$)~', $device);
+                if (
+                    count($device_parts) !== 2
+                    || !class_exists($device_parts[0])
```

Comment:
Calling `getById` would fail with a fatal error if the class is not a `CommonDBTM` class.
```suggestion
                    || !is_a($device_parts[0], \CommonDBTM::class, true)
```